src/test/java/seedu/address/logic/commands/FindCommandTest.java

```diff
@@ -1,83 +1,140 @@
-        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 3);
-        NameContainsKeywordsPredicate predicate = preparePredicate("Kurz Elle Kunz");
-        FindCommand command = new FindCommand(predicate);
-        expectedModel.updateFilteredPersonList(predicate);
+    public void execute_zeroKeywords_noTutorFound() {
+        String expectedMessage = String.format(MESSAGE_TUTORS_LISTED_OVERVIEW, 0);
+        NameContainsKeywordsPredicate predicate = null;
+        try {
+            predicate = preparePredicate("n/ ");
+        } catch (ParseException e) {
+            fail();
+        }
+        FindCommand command = new FindCommand(predicate, PersonType.TUTOR);
+        expectedModel.updateFilteredTutorList(predicate);
         assertCommandSuccess(command, model, expectedMessage, expectedModel);
-        assertEquals(Arrays.asList(CARL, ELLE, FIONA), model.getFilteredPersonList());
+        assertEquals(Collections.emptyList(), model.getFilteredTutorList());
+    }
+
+    @Test
+    public void execute_multipleStudentsFound() {
+        String expectedMessage = String.format(MESSAGE_STUDENTS_LISTED_OVERVIEW, 2);
+        try {
+            NameContainsKeywordsPredicate predicate = preparePredicate("n/john");
+            FindCommand command = new FindCommand(predicate, PersonType.STUDENT);
+            expectedSimilarNamesModel.updateFilteredStudentList(predicate);
+            assertCommandSuccess(command, similarNamesModel, expectedMessage, expectedSimilarNamesModel);
+            assertEquals(Arrays.asList(JOHN_P, JOHN_R), similarNamesModel.getFilteredStudentList());
+        } catch (ParseException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void execute_multipleTutorsFound() {
+        String expectedMessage = String.format(MESSAGE_TUTORS_LISTED_OVERVIEW, 2);
+        try {
+            NameContainsKeywordsPredicate predicate = preparePredicate("n/Don");
+            FindCommand command = new FindCommand(predicate, PersonType.TUTOR);
+            expectedSimilarNamesModel.updateFilteredTutorList(predicate);
+            assertCommandSuccess(command, similarNamesModel, expectedMessage, expectedSimilarNamesModel);
+            assertEquals(Arrays.asList(DON_A, DON_E), model.getFilteredTutorList());
+        } catch (ParseException e) {
+            fail();
+        }
     }
 
     /**
      * Parses {@code userInput} into a {@code NameContainsKeywordsPredicate}.
      */
-    private NameContainsKeywordsPredicate preparePredicate(String userInput) {
-        return new NameContainsKeywordsPredicate(Arrays.asList(userInput.split("\\s+")));
+    private NameContainsKeywordsPredicate preparePredicate(String userInput) throws ParseException {
+        Name name = FindCommandParserStub.extractName(userInput);
+        String[] nameList = new String[] {name.toString()};
+        return new NameContainsKeywordsPredicate(Arrays.asList(nameList));
```

Comment:
I think this should give the same result
```suggestion
        return new NameContainsKeywordsPredicate(Arrays.asList(name.toString()));
```